src/client/cli/commands/data/get.js

```diff
@@ -23,22 +25,59 @@ module.exports = {
 
   handler: subcommand((opts: {client: RestClient, objectIds: Array<string>, color: ?boolean, pretty: boolean}) => {
     const {client, objectIds, color, pretty} = opts
-    const printObject = objectPrinter(color, pretty)
+
+    if (objectIds.length === 1) {
+      return client.getData(objectIds[0])
+        .then(objectFormatter(color, pretty))
+        .then(output => process.stdout.write(output))
+    }
+
     return client.batchGetDataStream(objectIds)
-      .then(
-        stream => new Promise((resolve, reject) => {
-          stream.on('data', printObject)
-          stream.on('error', reject)
-          stream.on('end', resolve)
-        })
-      )
+      .then(stream => printStream(stream, objectIds, color, pretty))
+  })
+}
+
+function printStream (stream: ReadableStream, objectIds: Array<string>, color: ?boolean, pretty: boolean): Promise<void> {
+  const formatObject = objectFormatter(color, pretty)
+  const {stdout} = process
+  stdout.write('{')
+  if (pretty) {
+    stdout.write('\n')
+  }
+
+  return new Promise((resolve, reject) => {
+    const leftPad = pretty ? '  ' : ''
+    stream.on('data', result => {
+      const key = objectIds.pop()
+      const keyString = formatObject(key).trim()
+      const formatted = formatObject(result).trim()
+        .split('\n')
+        .map((s, i) => (i === 0) ? s : leftPad + s)
+        .join('\n')
+
+      stdout.write(`${leftPad}${keyString}: ${formatted}`)
+      if (objectIds.length > 0) {
+        stdout.write(',')
+      }
+      if (pretty) {
+        stdout.write('\n')
+      }
+    })
+    stream.on('error', (err) => {
+      stdout.write(`}\n`)
+      reject(err)
+    })
+    stream.on('end', () => {
```

Comment:
yes, the fetch request body stream will send the `end` event once you've consumed the whole thing.